src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -527,44 +527,50 @@ std::string ContextAttribute::toJson(bool isLastElement, bool types)
   {
     out = JSON_STR(name) + ":{" + JSON_STR("type") + ":" + JSON_STR(type) + "}"; 
   }
-  else if ((type == "") && (metadataVector.size() == 0))
+  else if ((renderMode == "values") || (renderMode == "keyValues"))
   {
+    out = (renderMode == "keyValues")? JSON_STR(name) + ":" : "";
+
     if (compoundValueP != NULL)
     {
       if (compoundValueP->isObject())
       {
-        out = JSON_STR(name) + ":{" + compoundValueP->toJson(true) + "}";
+        out += "{" + compoundValueP->toJson(true) + "}";
       }
       else if (compoundValueP->isVector())
       {
-        out = JSON_STR(name) + ":[" + compoundValueP->toJson(true) + "]";
+        out += "[" + compoundValueP->toJson(true) + "]";
       }
     }
     else if (valueType == orion::ValueTypeNumber)
     {
       char num[32];
-
       snprintf(num, sizeof(num), "%f", numberValue);
-      out = JSON_VALUE_NUMBER(name, num);
+      out += num;
     }
     else if (valueType == orion::ValueTypeString)
     {
-      out = JSON_VALUE(name, stringValue);
+      out += JSON_STR(stringValue);
     }
     else if (valueType == orion::ValueTypeBoolean)
     {
-      out = JSON_VALUE_BOOL(name, boolValue);
+      out += (boolValue == true)? "true" : "false";
     }
   }
-  else
+  else  // Render mode: normalized 
```

Comment:
Code comnets should be aligned with spec. In this particular case, I'm not sure if "normalized" will be a term used in the spec.

However, again, until things get clear I'm afraid cannot fix this.

NTC